### PR TITLE
Remove incorrect deprecation warning

### LINKED ISSUE #103444: Deprecation warning on lr_scheduler.step(num_steps)
### 🐛 Describe the bug

`step(num_steps)` produces a deprecation warning currently. However, there is a legitimate use case for this API in learning rate schedulers — if reloading a trained model and continuing to train, it is necessary to advance the number of steps inside the scheduler to match the current model state.

The scheduler does not provide an alternative way to advance the number of steps. And it is not possible to use `state_dict()`  + `load_state_dict()` because they also prevent changing the learning rate and other hyperparameters during transitions, unless the user manually changes the state dict, which is a hacky.

```
UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
```

### Versions

Versions of relevant libraries:
[pip3] torch==2.0.1

### PR DESCRIPTION
**Pull request by MicaelCarvalho**
Fixes #103444

The warning spams logs when resuming model training. Calling `lr_scheduler.step(num_steps)` to advance the scheduler to the current step count issues the warning.
A proposed alternative in the original issue is to use `state_dict`, however it also stores base learning rates. If the resumed model is a fine-tune with lower learning rate, re-loading the state dict overrides it with the original learning rate.
Which leaves us with the only option to remove the deprecation warning since this won't actually be deprecated.

1. Cannot use load_state_dict because this overrides the base learning rates and requires manual state dict manipulation
2. The deprecation of step(num_steps) has been canceled, per #103444, this warning has merged 3 years ago and isn't being pursued
3. This reverts part of changes from these PRs:
3.a https://github.com/pytorch/pytorch/pull/67954
3.b https://github.com/pytorch/pytorch/pull/26423
3.c https://github.com/pytorch/pytorch/pull/21460

### CODE REVIEW

**Comment by janeyx99** [comment]
Since we don't have active investment for LRScheduler today and thus will not be pushing through this deprecation anytime soon, I am okay with removing this warning. Let's rebase and get your change in.

**Review by janeyx99 — APPROVED** [review]

**Comment by janeyx99** [comment]
I meant to do a rebase 🤦🏻‍♀️

@pytorchbot rebase
